src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -56,7 +63,19 @@ public ModelMetadata getModelMetadata() {
      * @return length of model blob
      */
     public int getLength() {
-        return modelBlob.length;
+        if (modelBlob.get() == null) {
```

Comment:
may be use getter from 56

Reply:
do we have to mask length to zero if object is null? Why not let similar to other java objects like string/array?

Reply:
> may be use getter from 56
Will update

> do we have to mask length to zero if object is null? Why not let similar to other java objects like string/array?
GetLength is called in ModelCache to determine how large the model is. For model's whose modelBlob is null, we say the length is 0 so cache loading doesnt fail.